realm/realm-library/src/main/java/io/realm/internal/async/BadVersionException.java

```diff
@@ -24,7 +23,7 @@
  * state of the Realm which might be more up-to-date than the provided results or vice versa.
  */
 @Keep
-public class BadVersionException extends RealmException {
+public class BadVersionException extends Exception {
 
     public BadVersionException(String detailMessage) {
```

Comment:
Thinking again, I would rather leave a refactoring of this outside the scope of a patch PR. Also BadVersion might be appropriate after all as it could also happen if you try to handover to the wrong version = the version is reachable, but you just forgot to advance_read to it before doing the handover.